operators/multiclusterobservability/pkg/config/resources.go

```diff
@@ -0,0 +1,370 @@
+// Copyright (c) Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+// Licensed under the Apache License 2.0
+
+package config
+
+import (
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	mcoshared "github.com/stolostron/multicluster-observability-operator/operators/multiclusterobservability/api/shared"
+	observabilityv1beta2 "github.com/stolostron/multicluster-observability-operator/operators/multiclusterobservability/api/v1beta2"
+)
+
+const (
+	ResourceLimits                        = "limits"
+	ResourceRequests                      = "requests"
+	AnnotationMCOWithoutResourcesRequests = "mco-thanos-without-resources-requests"
+)
+
+// getDefaultResourceCPU returns the default resource CPU request for a particular o11y workload.
+func getDefaultResourceCPU(component string, tshirtSize observabilityv1beta2.TShirtSize) string {
+	switch component {
+	case ObservatoriumAPI:
+		return ObservatoriumAPICPURequest[tshirtSize]
+	case ThanosCompact:
+		return ThanosCompactCPURequest[tshirtSize]
+	case ThanosQuery:
+		return ThanosQueryCPURequest[tshirtSize]
+	case ThanosQueryFrontend:
+		return ThanosQueryFrontendCPURequest[tshirtSize]
+	case ThanosRule:
+		return ThanosRuleCPURequest[tshirtSize]
+	case ThanosReceive:
+		return ThanosReceiveCPURequest[tshirtSize]
+	case ThanosStoreShard:
+		return ThanosStoreCPURequest[tshirtSize]
+	case ThanosQueryFrontendMemcached, ThanosStoreMemcached:
+		return ThanosCachedCPURequest[tshirtSize]
+	case MemcachedExporter:
+		return MemcachedExporterCPURequest[tshirtSize]
+	case RBACQueryProxy:
+		return RBACQueryProxyCPURequest[tshirtSize]
+	case MetricsCollector:
+		return MetricsCollectorCPURequest[tshirtSize]
+	case Alertmanager:
+		return AlertmanagerCPURequest[tshirtSize]
+	case Grafana:
+		return GrafanaCPURequest[tshirtSize]
+	default:
+		return ""
+	}
+}
```

Comment:
This function's body looks like accessing a `map[string]ResourceSizeMap`, where the map key is the component's name.